src/Common/DateLUTImpl.cpp

```diff
@@ -291,15 +291,3 @@ namespace cctz_extension
 
     ZoneInfoSourceFactory zone_info_source_factory = custom_factory;
 }
-
-DateLUTImpl::Values DateLUTImpl::lutIndexByMonthSinceEpochStartsZeroIndexing(Int32 months) const
```

Comment:
Why is this being removed? What does DateLut has to do with iceberg?

Reply:
Removed in https://github.com/ClickHouse/ClickHouse/pull/77403, code in that PR totally refactored and this methods are not used anymore. PR added partition pruning for Iceberg.
https://github.com/ClickHouse/ClickHouse/pull/77403/files#r1998468379

Reply:
PR makes a refactoring in Iceberg code, and PR Iceberg metadata cache changes the same code later, so I cherry-picked it to get less conflicts.